autogpt_platform/backend/backend/blocks/llm.py

```diff
@@ -973,31 +995,42 @@ def validate_response(parsed: object) -> str | None:
 
                 if input_data.expected_format:
                     try:
-                        response_obj = json.loads(response_text)
-                    except JSONDecodeError as json_error:
+                        response_obj = None
+                        if input_data.force_json_output:
+                            # Handle pure JSON responses
+                            with contextlib.suppress(JSONDecodeError):
+                                response_obj = json.loads(response_text)
+
+                        if response_obj is None:
+                            if output_tag_start not in response_text:
+                                raise ValueError(
+                                    "Response does not contain the expected "
+                                    f"{output_tag_start}...</json_output> block."
+                                )
+                            json_output = (
+                                response_text.split(output_tag_start, 1)[1]
+                                .rsplit("</json_output>", 1)[0]
+                                .strip()
+                            )
+                            response_obj = json.loads(json_output)
+                    except (ValueError, JSONDecodeError) as parse_error:
                         prompt.append({"role": "assistant", "content": response_text})
 
-                        indented_json_error = str(json_error).replace("\n", "\n|")
+                        indented_parse_error = str(parse_error).replace("\n", "\n|")
                         error_feedback_message = trim_prompt(
                             f"""
-                            |Your previous response could not be parsed as valid JSON:
+                            |Your previous response did not contain a parseable JSON {output_type}:
                             |
-                            |{indented_json_error}
+                            |{indented_parse_error}
                             |
-                            |Please provide a valid JSON response that matches the expected format.
+                            |Please provide a {output_tag_start}...</json_output> block containing a valid JSON {output_type} that matches the expected format.
                         """
                         )
                         prompt.append(
                             {"role": "user", "content": error_feedback_message}
                         )
                         continue
 
-                    if input_data.list_result and isinstance(response_obj, dict):
```

Comment:
Do we retain the behavior of forcing the result into a list when this is enabled?